lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_firebase_test.rb

```diff
@@ -0,0 +1,168 @@
+          version: params[:version],
+          locale: params[:locale],
+          orientation: params[:orientation]
+        )
+
+        result = test_runner.run_tests(
+          apk_path: params[:apk_path],
+          test_apk_path: params[:test_apk_path],
+          device: device,
+          type: params[:type]
+        )
+
+        # Download all of the outputs from the job to the local machine
+        test_runner.download_result_files(
+          result: result,
+          destination: test_dir,
+          project_id: params[:project_id],
+          key_file_path: params[:key_file]
+        )
+
+        FastlaneCore::UI.test_failure! "Firebase Tests failed – more information can be found at #{result.more_details_url}" unless result.success?
+
+        UI.success 'Firebase Tests Complete'
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Runs the specified tests in Firebase Test Lab'
+      end
+
+      def self.details
+        description
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :project_id,
+            # `env_name` comes from the Google Cloud default: https://cloud.google.com/functions/docs/configuring/env-var
+            env_name: 'GCP_PROJECT',
+            description: 'The Project ID to test in',
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :key_file,
+            description: 'The key file used to authorize with Google Cloud',
+            type: String,
+            verify_block: proc do |value|
+              UI.user_error!("Invalid key file path: #{value}") unless File.exist?(value)
+            end
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :apk_path,
+            description: 'Path to the application APK on the local machine',
+            type: String,
+            verify_block: proc do |value|
+              UI.user_error!("Invalid application APK: #{value}") unless File.exist?(value)
+            end
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :test_apk_path,
+            description: 'Path to the test bundle APK on the local machine',
+            type: String,
+            verify_block: proc do |value|
+              UI.user_error!("Invalid test APK: #{value}") unless File.exist?(value)
+            end
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :model,
+            description: 'The device model to use to run the test',
+            type: String,
+            verify_block: proc do |value|
+              FirebaseTestRunner.verify_has_gcloud_binary!
+              model_names = Fastlane::FirebaseDevice.valid_model_names
+              UI.user_error!("Invalid Model Name: #{value}. Valid Model Names: #{model_names}") unless model_names.include?(value)
```

Comment:
The user_error doesn't seem to show (and thus list the available models) when run on CI and we pass an invalid model — see example [here](https://buildkite.com/automattic/woocommerce-android/builds/4205#0181c9fe-56b2-48e0-b9ee-1cd2ebe0eed8). It just says the parameter has been passed invalid value but the message above doesn't show on the logs, so that is not super helpful 😒 

```
  | [2022-07-04T16:30:17Z] [16:30:17]: ▸ BUILD SUCCESSFUL in 12m 56s
  | [2022-07-04T16:30:17Z] [16:30:17]: ▸ 94 actionable tasks: 86 executed, 8 from cache
  | [2022-07-04T16:30:17Z] [16:30:17]: ▸ Publishing build scan...
  | [2022-07-04T16:30:17Z] [16:30:17]: ▸ https://gradle.a8c.com/s/wz445ksdkrrek
  | [2022-07-04T16:30:19Z] [16:30:19]: Error setting value 'Nexus5' for option 'model'
  | [2022-07-04T16:30:19Z] [16:30:19]: You passed invalid parameters to 'android_firebase_test'.
  | [2022-07-04T16:30:19Z] [16:30:19]: Check out the error below and available options by running `fastlane action android_firebase_test`
  | [2022-07-04T16:30:19Z] +-------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
  | [2022-07-04T16:30:19Z] \|                                                                                                                                                                                                       Lane Context                                                                                                                                                                                                       \|
  | [2022-07-04T16:30:19Z] +-------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
  | [2022-07-04T16:30:19Z] \| DEFAULT_PLATFORM                    \| android                                                                                                                                                                                                                                                                                                                                                                            \|
  | [2022-07-04T16:30:19Z] \| PLATFORM_NAME                       \| android                                                                                                                                                                                                                                                                                                                                                                            \|
  | [2022-07-04T16:30:19Z] \| LANE_NAME                           \| android build_and_instrumented_test                                                                                                                                                                                                                                                                                                                                                \|
  | [2022-07-04T16:30:19Z] \| GRADLE_ALL_APK_OUTPUT_PATHS         \| ["/var/lib/buildkite-agent/builds/ci-android-i-04a64e2244229b61c-1/automattic/woocommerce-android/WooCommerce/build/outputs/apk/vanilla/debug/WooCommerce-vanilla-debug.apk", "/var/lib/buildkite-agent/builds/ci-android-i-04a64e2244229b61c-1/automattic/woocommerce-android/WooCommerce/build/outputs/apk/androidTest/vanilla/debug/WooCommerce-vanilla-debug-androidTest.apk"] \|
  | [2022-07-04T16:30:19Z] \| GRADLE_ALL_AAB_OUTPUT_PATHS         \| []                                                                                                                                                                                                                                                                                                                                                                                 \|
  | [2022-07-04T16:30:19Z] \| GRADLE_ALL_OUTPUT_JSON_OUTPUT_PATHS \| ["/var/lib/buildkite-agent/builds/ci-android-i-04a64e2244229b61c-1/automattic/woocommerce-android/WooCommerce/build/outputs/apk/vanilla/debug/output-metadata.json", "/var/lib/buildkite-agent/builds/ci-android-i-04a64e2244229b61c-1/automattic/woocommerce-android/WooCommerce/build/outputs/apk/androidTest/vanilla/debug/output-metadata.json"]                               \|
  | [2022-07-04T16:30:19Z] \| GRADLE_ALL_MAPPING_TXT_OUTPUT_PATHS \| []                                                                                                                                                                                                                                                                                                                                                                                 \|
  | [2022-07-04T16:30:19Z] \| GRADLE_APK_OUTPUT_PATH              \| /var/lib/buildkite-agent/builds/ci-android-i-04a64e2244229b61c-1/automattic/woocommerce-android/WooCommerce/build/outputs/apk/androidTest/vanilla/debug/WooCommerce-vanilla-debug-androidTest.apk                                                                                                                                                                                  \|
  | [2022-07-04T16:30:19Z] \| GRADLE_OUTPUT_JSON_OUTPUT_PATH      \| /var/lib/buildkite-agent/builds/ci-android-i-04a64e2244229b61c-1/automattic/woocommerce-android/WooCommerce/build/outputs/apk/androidTest/vanilla/debug/output-metadata.json                                                                                                                                                                                                       \|
  | [2022-07-04T16:30:19Z] +-------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
  | [2022-07-04T16:30:19Z] [16:30:19]: Shell command exited with exit status 1 instead of 0.
  | [2022-07-04T16:30:20Z]
  | [2022-07-04T16:30:20Z] +------+------------------------------------------------------+-------------+
  | [2022-07-04T16:30:20Z] \|                             fastlane summary                              \|
  | [2022-07-04T16:30:20Z] +------+------------------------------------------------------+-------------+
  | [2022-07-04T16:30:20Z] \| Step \| Action                                               \| Time (in s) \|
  | [2022-07-04T16:30:20Z] +------+------------------------------------------------------+-------------+
  | [2022-07-04T16:30:20Z] \| 1    \| default_platform                                     \| 0           \|
  | [2022-07-04T16:30:20Z] \| 2    \| is_ci                                                \| 0           \|
  | [2022-07-04T16:30:20Z] \| 3    \| assembleVanillaDebug assembleVanillaDebugAndroidTest \| 777         \|
  | [2022-07-04T16:30:20Z] +------+------------------------------------------------------+-------------+
  | [2022-07-04T16:30:20Z]
  | [2022-07-04T16:30:20Z] [16:30:20]: fastlane finished with errors
  | [2022-07-04T16:30:20Z]
  | [2022-07-04T16:30:20Z] [!] Shell command exited with exit status 1 instead of 0.
  | [2022-07-04T16:30:20Z] 🚨 Error: The command exited with status 1
  | [2022-07-04T16:30:20Z] user command error: exit status 1
```